src/index.ts

```diff
@@ -16,6 +16,58 @@ const rl = createInterface({
   output: process.stdout
 })
 
+function connectToBootnode (node: any, bootNodeAddr: string) {
+  return new Promise((resolve, reject) => {
+    async.parallel(
+      [
+        (cb: () => void) =>
+          node.dial(new Multiaddr(bootNodeAddr), cb),
+        // Set up of the cons might take time
+        (cb: () => void) => setTimeout(cb, 300)
+      ],
+      (err: Error | null | undefined) => {
+        if (err) {
+          reject(err)
+        }
+        logger.info('Connection Successful')
+
+        if (myArgs.chatClient) {
+          const chatClient: CommandLineChat = new CommandLineChat(node)
+          chatClient.init()
+        }
+        resolve()
+      }
+    )
+  })
+}
+
+async function processBootnodes (node: any, bootNodeAddresses: Array<string>) {
+  let connected = false
+  const iterator = bootNodeAddresses[Symbol.iterator]()
+  let count = 0
+  let bootNodeAddr = iterator.next().value
+
+  while (count < bootNodeAddresses.length && !connected) {
```

Comment:
If you want to mimic for cycle then use the `for` keyword :) . `for (let i = 0; i < bootNodeAddresses.length && !connected; ++i)`

Reply:
so C like 😄

Reply:
yep :)